pkg/controller/versions/strategy_v2_1.go

```diff
@@ -266,6 +271,62 @@ func (v *versionStrategyV2_1) Render(ctx context.Context, cr *common.ControllerR
 		return nil, fmt.Errorf("Could not set field status.lastAppliedConfiguration.istio.istio_cni.istio_cni_network: %v", err)
 	}
 
+	if smcp.Spec.ControlPlaneMode != nil {
+		if smcp.Spec.ControlPlaneMode.Enabled != nil {
+			err = spec.Istio.SetField("istioDiscovery.enabled", *smcp.Spec.ControlPlaneMode.Enabled)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			err = spec.Istio.SetField("istioDiscovery.enabled", true)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if smcp.Spec.Telemetry != nil {
+		if smcp.Spec.Telemetry.Enabled != nil {
```

Comment:
why add the Enabled field to Telemetry? can't you just set `telemetry.common.enabled` to the value of  `smcp.Spec.ControlPlaneMode.Enabled` like above for istioDiscovery?

Reply:
We wanted the possibility to switch off all telemetry resources and this was not necessarily a split control plane related situation
Everything can be disabled from the telemetry stack except this chart. Maistra only needs it when at least one component is enabled from the telemetry addons.

Reply:
See my comment above.  You should be able to disable telemetry by setting telemetry.type=None

Reply:
The `telemetry-common` chart will be installed by default:
https://github.com/maistra/istio-operator/blob/fdcaa2af68e1c1950391f747e5f5e6c519eeb038/pkg/controller/versions/strategy_v2_1.go#L43-L45

Reply:
@dgn @rcernich our insight is that the charts which has empty `enabledField` setting will always be rendered and deployed at least partially (based on how other values are set). Ref: https://github.com/mtturcsik/istio-operator/blob/7faff8fc6c1d005f35639ed4f38bc5f13ab26061/pkg/controller/versions/strategy_v2_1.go#L471

Isn't this the intended behavior?